app/controllers/campaigns_controller.rb

```diff
@@ -16,6 +16,9 @@ def create
 
   def show
     @campaign = Campaign.find params['id']
+    if @campaign.active == false
+      redirect_to :campaigns, notice: 'The campaign you requested has been deactivated.'
```

Comment:
I think accessing inactive pages should be different for logged in campaigners vs. members. Campaigners could preferrably have the option to view stats for and to re-activate old campaigns, whereas members should get a 404/redirect.